crates/net/downloaders/src/file_client.rs

```diff
@@ -49,6 +47,9 @@ pub struct FileClient {
 
     /// The buffered bodies retrieved when fetching new headers.
     bodies: HashMap<BlockHash, BlockBody>,
+
+    /// The codec used to decode the file.
+    codec: T,
```

Comment:
right so we can remove this field completely...and the generic on file client...would you be ok with doing this last thing or should I?

Reply:
Yeah I will try to do it

Reply:
I'm not really sure how to remove the generic? Isn't it the part of this pull request, or should I use phantom field for the type T?

Reply:
the pr makes it apparent that the generic isn't needed: the codec is passed as arg instead, there is no need to save it in the `FileClient` type (the unused code warning is rust compilers way of hinting about this!).

the current design of how `FileClient` and `ChunkedFileReader` are connected, isn't great rn https://github.com/paradigmxyz/reth/issues/7650, apologies for the confusion!